### PR TITLE
CMR-4093 Update access-control docs to use EUI

### PR DESCRIPTION
**Pull request by oubiwann**
In order to support EUI in `access-control`, this change also had to do the following:
 * Generalize `search-app` template code and static generation
 * Move generalized code into `common-app-lib`
 * Create EUI-ized `access-control` templates, pages, data, and routes
 * Update all routes to work well together when running locally (i.e., with no app prefix in the URL path)
 * `access-control` routes code was missing unit tests, so I added these

### CODE REVIEW

File: access-control-app/src/cmr/access_control/routes.clj [hunk 1]
```diff
@@ -0,0 +1,39 @@
+(ns cmr.access-control.routes
+  "This namespace is the one responsible for combining the routes intended for
+  use by libraries and applications (API routes) and the routes intended for
+  human, browser, or web crawler consumption (site routes). It also provides
+  a place for adding routes that apply to both (e.g., robots.txt, should we
+  ever need that for access-control)."
+  (:require
+   [cheshire.core :as json]
```
Comment (indiejames) [inline access-control-app/src/cmr/access_control/routes.clj]: This require doesn't seemed to be needed for this namespace.

File: access-control-app/src/cmr/access_control/routes.clj [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+   [cmr.access-control.api.routes :as api-routes]
+   [cmr.access-control.site.routes :as site-routes]
+   [cmr.acl.core :as acl]
+   [cmr.common-app.api-docs :as api-docs]
```
Comment (indiejames) [inline access-control-app/src/cmr/access_control/routes.clj]: Don't see this used here.

File: access-control-app/src/cmr/access_control/site/routes.clj [hunk 3]
```diff
@@ -0,0 +1,59 @@
+(ns cmr.access-control.site.routes
+  "Defines the HTTP URL routes for the access-control web site."
+  (:require
+   [cmr.access-control.site.pages :as pages]
+   [cmr.common-app.api-docs :as api-docs]
+   [cmr.common-app.site.data :as common-data]
```
Comment (indiejames) [inline access-control-app/src/cmr/access_control/site/routes.clj]: I don't see this or the next require used here.

File: access-control-app/src/cmr/access_control/routes.clj (hunk 1 above)
Comment (oubiwann) [inline access-control-app/src/cmr/access_control/routes.clj]: Indeed -- fixed.

File: access-control-app/src/cmr/access_control/routes.clj (hunk 2 above)
Comment (oubiwann) [inline access-control-app/src/cmr/access_control/routes.clj]: Fixed.

File: access-control-app/docs/site.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+### Table of Contents
+
+  * [Routes](#routes)
+  * [Static Assets and Content](#static-assets-and-content)
+  * [Redirects](#redirects)
+
+### <a name="routes"></a> Routes
+
+The CMR Search Site defines the following application routes, relative to the base CMR Search URL. These resources at the URLs are generated dynamically using page templates (cached).
```
Comment (ygliuvt) [inline access-control-app/docs/site.md]: Looks like this is copied from CMR search. Shouldn't we change CMR Search to CMR Access Control through out the document?

Comment (oubiwann) [inline access-control-app/docs/site.md]: Good catch -- I'll hit that now.

Comment (oubiwann) [inline access-control-app/docs/site.md]: Fixed.

**Pull request merged**